packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.tsx

```diff
@@ -0,0 +1,149 @@
+  eventData: Record<string, unknown>;
+};
+
+/*
+  TODO: These types are temporary until we have the proper
+  ones generated from the Settings schema
+*/
+export type TPermissionGroup = {
+  name: string;
+  oAuthScopes: string[];
+};
+
+export type TCustomView = {
+  id: string;
+  defaultLabel: string;
+  labelAllLocales: Record<string, string>;
+  url: string;
+  type: 'CustomPanel' | 'CustomTab';
+  typeConfig?: {
+    size?: 'SMALL' | 'LARGE';
+  };
+  locators: string[];
+};
+
+type TCustomViewLoaderProps = {
+  customView: TCustomView;
+  onClose: () => void;
+};
+
+const isIframeReady = (iFrameElementRef: HTMLIFrameElement) => {
+  try {
+    return iFrameElementRef?.contentWindow?.document.readyState === 'complete';
+  } catch {
+    // Trying to access the contentWindow of a cross-origin iFrame will throw an error.
+    // We are not supposed to even get here because the iFrame must use
+    // a URL from our very same domain (the custom view is proxied through our http-proxy service).
+    return false;
+  }
+};
+
+const CustomPanelIframe = styled.iframe`
+  height: 100%;
+  width: 100%;
+  border: none;
+`;
+
+function CustomViewLoader(props: TCustomViewLoaderProps) {
+  const iFrameElementRef = useRef<HTMLIFrameElement>(null);
+  const appContext = useApplicationContext();
+  const iFrameCommunicationChannel = useRef(new MessageChannel());
+  const showNotification = useShowNotification();
+  const intl = useIntl();
+  const panelSize = (props.customView.typeConfig?.size?.toLocaleLowerCase() ||
+    'large') as Lowercase<'SMALL' | 'LARGE'>;
+
+  const messageFromIFrameHandler = useCallback((event: MessageEvent) => {
+    if (event.data.origin === window.location.origin) {
+      console.log('message received from iframe port: ', event);
+    }
+  }, []);
+
+  // onLoad handler is called from the iFrame even where the URL is not valid
+  // (blocked by CORS, 404, etc.) so we need to make sure the iFrame is ready
+  const onLoadSuccessHandler = useCallback(() => {
+    // Show error and block if the iFrame is not ready
+    // (error loading it)
+    if (!isIframeReady(iFrameElementRef.current!)) {
+      showNotification({
+        domain: DOMAINS.PAGE,
+        kind: NOTIFICATION_KINDS_PAGE.error,
+        text: intl.formatMessage(messages.loadError),
+      });
+      return;
+    }
+
+    // Listen for messages from the iFrame
+    iFrameCommunicationChannel.current.port1.onmessage =
+      messageFromIFrameHandler;
+
+    // Transfer port2 to the iFrame so it can send messages back privately
+    iFrameElementRef.current?.contentWindow?.postMessage(
+      'mc_init',
+      window.location.href,
+      [iFrameCommunicationChannel.current.port2]
+    );
+
+    // Send the initialization message to the iFrame
+    iFrameCommunicationChannel.current.port1.postMessage({
+      source: 'mc-host-application',
+      destination: `custom-view-${props.customView.id}`,
+      eventName: 'hostAcknowledgesIframeLoaded',
```

Comment:
Is this a standard event name or something we need to define?

If the latter, maybe we can put these constants into a constants file/object, as I see a bunch of other cases here like `mc_init`, `mc-host-application`, etc.
Also the casing seems to be a bit inconsistent. Is this on purpose?